x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/action/GetGlobalCheckpointsAction.java

```diff
@@ -157,89 +161,178 @@ public TransportAction(
         @Override
         protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
             final ClusterState state = clusterService.state();
-            final Index index = resolver.concreteSingleIndex(state, request);
-            final IndexMetadata indexMetadata = state.getMetadata().index(index);
 
-            if (indexMetadata == null) {
-                // Index not found
-                listener.onFailure(new IndexNotFoundException(request.index));
+            final Index index;
+            try {
+                index = resolver.concreteSingleIndex(state, request);
+            } catch (IndexNotFoundException e) {
+                handleIndexNotReady(request, listener);
```

Comment:
Should we fail out immediately if there are no `wait_for_xxx` options specified? See also comment below.